### ISSUE TITLE
Crashes on long single row png (36000x400)

### ISSUE THREAD

**Issue by asyschikov** [Audiveris/audiveris#173]
I have a long png file (36000x400) which crashes with the following exception
```
WARN  [out]               GridBuilder 149  | Error in GridBuilder: java.lang.RuntimeException: Illegal run start -30610
java.lang.RuntimeException: Illegal run start -30610
	at org.audiveris.omr.run.RunTable.addRun(RunTable.java:199)
	at org.audiveris.omr.run.RunTable.addRun(RunTable.java:179)
	at org.audiveris.omr.run.RunTable.purge(RunTable.java:1221)
	at org.audiveris.omr.sheet.grid.LinesRetriever.buildHorizontalLag(LinesRetriever.java:197)
	at org.audiveris.omr.sheet.grid.GridBuilder.buildAllLags(GridBuilder.java:201)
	at org.audiveris.omr.sheet.grid.GridBuilder.buildInfo(GridBuilder.java:109)
	at org.audiveris.omr.sheet.grid.GridStep.doit(GridStep.java:94)
	at org.audiveris.omr.step.Step.doit(Step.java:168)
	at org.audiveris.omr.sheet.BasicStub$3.call(BasicStub.java:814)
	at org.audiveris.omr.sheet.BasicStub$3.call(BasicStub.java:802)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
WARN  [out]                   Picture 904  | No system lines to build NO_STAFF buffer
WARN  [out]                   Picture 904  | No system lines to build NO_STAFF buffer
WARN  [out]                   Picture 904  | No system lines to build NO_STAFF buffer
WARN  [out]                 BasicStub 841  | Error in GRID java.util.concurrent.ExecutionException: java.lang.NullPointerException
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
	at org.audiveris.omr.sheet.BasicStub.doOneStep(BasicStub.java:824)
	at org.audiveris.omr.sheet.BasicStub.reachStep(BasicStub.java:564)
	at org.audiveris.omr.sheet.BasicBook.reachBookStep(BasicBook.java:1063)
	at org.audiveris.omr.sheet.BasicBook.transcribe(BasicBook.java:1446)
	at org.audiveris.omr.sheet.BasicBook.export(BasicBook.java:558)
	at org.audiveris.omr.CLI$ProcessingTask.processBook(CLI.java:785)
	at org.audiveris.omr.CLI$CliTask.call(CLI.java:344)
	at org.audiveris.omr.CLI$CliTask.call(CLI.java:307)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
	at org.audiveris.omr.sheet.SystemManager.dispatchHorizontalSections(SystemManager.java:216)
	at org.audiveris.omr.sheet.SystemManager.populateSystems(SystemManager.java:448)
	at org.audiveris.omr.sheet.grid.StaffLineCleaner.process(StaffLineCleaner.java:101)
	at org.audiveris.omr.sheet.grid.GridStep.doit(GridStep.java:97)
	at org.audiveris.omr.step.Step.doit(Step.java:168)
	at org.audiveris.omr.sheet.BasicStub$3.call(BasicStub.java:814)
	at org.audiveris.omr.sheet.BasicStub$3.call(BasicStub.java:802)
	... 4 common frames omitted
```

I am pretty sure that the size has to do with it, because I cropped it to 75% and it worked ok. I can provide the PNG via a private channel because it is a copyrighted material.

**Comment by hbitteur** [comment]
I found 3 problems and was able to resolve only the first 2 ones:

- Internally, Audiveris implements binarized images using RLE (Run Length Encoding), with the run length coded on a "short". A short number covers range -32768..**32767**. Your image is **36801** pixels wide, hence the first long (black or white) horizontal run will lead to overflow. I fixed that by discarding shorts for ints, which is too bad for the memory resource, but the bug is fixed.
- There is no white margin around the image. At location (x:8180, y:444), a beam touches the image limits. I had to strengthen the code to avoid  an ArrayOutOfBoundsException. Bug fixed.
- Then the OCR fails because external Tesseract program fails with an exit value of -1. Just in case, I tried to dynamically add some white margin around the image before handing it to Tesseract, but it failed again.

Additional work is needed. Perhaps trying to launch Tesseract on your image (totally outside of Audiveris) might shed some light on its behavior.

**Comment by hbitteur** [comment]
Bug fixes are in latest commits (5ed19bf6830dc925ef3e0347f0a910b296c1937c, 0c3a0a5d004b7023c1df110adcb165b2b90b98a2 and 0b61ddee333b50ee2abce62d8692d0f5bacc8b5a)

Although I could not fix the Tesseract problem on your image, only the TEXTS step fails, which means that you can safely let the processing go now.
There are just a few manual corrections to perform at the end:

- A chord of whole heads at measure 56
- One slur at measure 57
- The word "Swing" which is correctly OCR'ed when "text" shape is manually assigned to it.

**Comment by asyschikov** [comment]
Great, thanks for the update!
